tests/functionnal/SLM.php

```diff
@@ -399,4 +622,158 @@ public function testInternalTtrComputation() {
+            'type'               => \SLM::TTR,
+            'number_time'        => 1,
+            'definition_time'    => 'month',
+            'end_of_working_day' => false,
+         ]
+      );
+      $this->integer($ola_ttr_id)->isGreaterThan(0);
+
+      $ola_ttr = new \OLA();
+      $ola_tto_id = $ola_ttr->add(
+         [
+            'slms_id'            => $slm_id,
+            'name'               => 'Test TTO OLA',
+            'type'               => \SLM::TTO,
+            'number_time'        => 1,
+            'definition_time'    => 'month',
+            'end_of_working_day' => false,
+         ]
+      );
+      $this->integer($ola_tto_id)->isGreaterThan(0);
+
+      $sla_tto = new \SLA();
+      $sla_ttr_id = $sla_tto->add(
+         [
+            'slms_id'            => $slm_id,
+            'name'               => 'Test TTR SLA',
+            'type'               => \SLM::TTR,
+            'number_time'        => 1,
+            'definition_time'    => 'month',
+            'end_of_working_day' => false,
+         ]
+      );
+      $this->integer($sla_ttr_id)->isGreaterThan(0);
+
+      $sla_ttr = new \SLA();
+      $sla_tto_id = $sla_ttr->add(
+         [
+            'slms_id'            => $slm_id,
+            'name'               => 'Test TTO SLA',
+            'type'               => \SLM::TTO,
+            'number_time'        => 1,
+            'definition_time'    => 'month',
+            'end_of_working_day' => false,
+         ]
+      );
+      $this->integer($sla_tto_id)->isGreaterThan(0);
+
+      // Create ticket with SLA/OLA TTO/TTR to test computation based on SLA OLA
+      $createtime = time();
+      $ticket = new \Ticket();
+      $ticket_id = $ticket->add(
+         [
+            'name'    => 'Test Ticket',
+            'content' => 'Ticket for TTR OLA test on create',
+            'olas_id_ttr' => $ola_ttr_id,
+            'olas_id_tto' => $ola_tto_id,
+            'slas_id_ttr' => $sla_ttr_id,
+            'slas_id_tto' => $sla_tto_id,
+         ]
+      );
+      $this->integer($ticket_id)->isGreaterThan(0);
+
+      $after1month = date('Y-m-d H:i:s', strtotime($ticket->fields['date'].'+1 month'));
+      $this->boolean($ticket->getFromDB($ticket_id))->isTrue();
+
+      //check computed data from SLA / OLA
+      $this->integer((int)$ticket->fields['olas_id_ttr'])->isEqualTo($ola_ttr_id);
+      $this->variable($ticket->fields['internal_time_to_resolve'])->isEqualTo($after1month);
+
+      $this->integer((int)$ticket->fields['olas_id_tto'])->isEqualTo($ola_tto_id);
+      $this->variable($ticket->fields['internal_time_to_own'])->isEqualTo($after1month);
+
+      $this->integer((int)$ticket->fields['slas_id_ttr'])->isEqualTo($sla_ttr_id);
+      $this->variable($ticket->fields['time_to_resolve'])->isEqualTo($after1month);
+
+      $this->integer((int)$ticket->fields['slas_id_tto'])->isEqualTo($sla_tto_id);
+      $this->variable($ticket->fields['time_to_own'])->isEqualTo($after1month);
```

Comment:
I would test different dates for each field. In deed, if there is a mixup between OLA/SLA TTR/TTO, this test will not detect it.

Reply:
Fixed